tests/run-make/pointer-auth-link-with-c-lto-clang/rmake.rs

```diff
@@ -32,5 +35,15 @@ fn main() {
         .input("test.rs")
         .output("test.bin")
         .run();
+
+    // Check that both a-key and b-key pac-ret survived LTO
```

Comment:
Question: is there any background / links for this? I'm fine with the change, but it's even better if there's any kind of context to read.

Reply:
For sure, this is the context:
https://github.com/llvm/llvm-project/commit/1782810b8440144a0141c24192acbaeb55a1545d

TLDR: there was a bug in LLVM where during the LTO step instructions like "pacibsp" would get replaced with "paciasp" due to LLVM module flag merging behaviour. As currently written the assembly output should have no PAuth on library functions, `paciasp` at the start of main and `pacibsp` at the start of `foo`.
 
It is worrying that the test is failing, I'll try to reproduce it locally.

Reply:
I agree, the test failing is... suspicious lol.

Reply:
Also it might help if you do something like `let clang_version = clang().arg("-v").run();` and then print that to double-check what clang version CI has.

Reply:
Okay this is very funny
```
00000000000677a4 <test::main>:
   677a4: d503233f      paciasp
   677a8: d50323bf      autiasp
   677ac: d65f03c0      ret
```
The test fails because the compiler optimises away the entire call to `foo` and its pac instructions along with it. I'll fix it..

Reply:
Right... That can explain it lol. `black_box` to the rescue.